src/main/java/org/commonjava/indy/service/archive/controller/ArchiveController.java

```diff
@@ -203,6 +272,64 @@ public void deleteArchive( final String buildConfigId )
         logger.info( "Historical archive for build config id: {} is deleted.", buildConfigId );
     }
 
+    public void deleteArchiveWithChecksum( final String buildConfigId, final String checksum )
+            throws IOException
+    {
+        logger.info( "Start to delete archive with checksum validation, buildConfigId {}, checksum {}", buildConfigId,
+                     checksum );
+        File zip = new File( archiveDir, buildConfigId + ARCHIVE_SUFFIX );
+        if ( !zip.exists() )
+        {
+            return;
+        }
+
+        try (FileChannel channel = new FileInputStream( zip ).getChannel())
+        {
+            MessageDigest digest = MessageDigest.getInstance( SHA_256 );
+            long position = 0;
+            long size = channel.size();
+
+            while ( position < size )
+            {
+                long remaining = size - position;
+                long currentBlock = Math.min( remaining, BOLCK_SIZE );
+                MappedByteBuffer buffer = channel.map( FileChannel.MapMode.READ_ONLY, position, currentBlock );
+                digest.update( buffer );
+                position += currentBlock;
```

Comment:
Is there any 3rd lib for this? like e.g.: commons-codec .

Reply:
@sswguo done in new commit.